### PR TITLE
VcPkg: On Windows, also install WinHTTP by default

### PR DESCRIPTION
**Pull request by antkmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]
Can we have comments in the CONTROL file?

It would be nice to have a comment about why we are adding the http feature. Or maybe the Description is enough.

Thank you

**Comment by antkmsft** [comment]
> Can we have comments in the CONTROL file?

I haven't found any comments. But I'll check with Billy later.
If comments are possible, then the other thing we will add is the license and copyright at the top.

CONTROL is supposed to be a subset of some Debian format, and I think I have tried several comment formats before, and failed. And vcpkg documentation does not mention any comment ability. Billy would know ultimately.

File: sdk/core/azure-core/vcpkg/CONTROL [hunk 1]
```diff
@@ -5,9 +5,14 @@ Description: Microsoft Azure Core SDK for C++
   This library provides shared primitives, abstractions, and helpers for modern Azure SDK client libraries written in the C++.
 Homepage: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/core/azure-core
 Supports: !uwp
-Default-Features: curl
+Default-Features: http
+
+Feature: http
+Build-Depends: azure-core-cpp[core, curl], azure-core-cpp[core, curl, winhttp] (windows)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/vcpkg/CONTROL:11]: What is `core`?

File: sdk/core/azure-core/vcpkg/CONTROL [hunk 2]
```diff
@@ -5,9 +5,14 @@ Description: Microsoft Azure Core SDK for C++
...
+Description: Build all HTTP transport implementations, depending on the platform
```
Comment (ahsonkhan) [inline sdk/core/azure-core/vcpkg/CONTROL:12]:
```suggestion
Description: Build all HTTP transport implementations, depending on the platform.
```

**Review by ahsonkhan — APPROVED** [review]

File: sdk/core/azure-core/vcpkg/CONTROL (hunk 2 above)
Comment (antkmsft) [reply sdk/core/azure-core/vcpkg/CONTROL:12]: I looked at a few packages randomly, including CURL, other packages don't end feature descriptions with `.`

File: sdk/core/azure-core/vcpkg/CONTROL (hunk 1 above)
Comment (antkmsft) [reply sdk/core/azure-core/vcpkg/CONTROL:11]: It is the built-in feature in vcpkg, every package has it. Core, in VcPkg terms, is the package without any features.

File: sdk/core/azure-core/vcpkg/CONTROL (hunk 2 above)
Comment (antkmsft) [reply sdk/core/azure-core/vcpkg/CONTROL:12]: Examples: https://github.com/microsoft/vcpkg/blob/master/ports/curl/CONTROL, https://github.com/microsoft/vcpkg/blob/master/ports/aws-sdk-cpp/CONTROL

**Pull request merged**